src/patch_parser.py

```diff
@@ -414,105 +422,96 @@ def _does_match_end_of_super_chunk(lines: InputSource) -> bool:
-                    gerrit_new_line,
-                    PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
-                                          side='',
-                                          offset=(gerrit_new_line - lines.line_number() - 1)))
-    return (gerrit_orig_line,
-            gerrit_new_line,
-            PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
-                                  side='',
-                                  offset=(gerrit_new_line - lines.line_number())))
+        parser_state.gerrit_orig_line += 1
+        parser_state.gerrit_new_line += 1
+        if previous[0] == '-':  # TODO: maybe add check to see if it is actually a modified one though I think this
+            # situation only applies when it actually is modified
+            return PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
+                                         side='',
+                                         offset=(parser_state.gerrit_new_line - lines.line_number() - 1))
+    return PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
+                                 side='',
+                                 offset=(parser_state.gerrit_new_line - lines.line_number()))
 
 
 def _parse_patch_file_removed_chunk(
         lines: InputSource,
-        gerrit_orig_line: int,
-        gerrit_new_line: int) -> Tuple[int, int, PatchFileChunkLineMap]:
+        parser_state: HunkParserState) -> PatchFileChunkLineMap:
     in_start = lines.line_number()
     while lines[0] and lines[0][0] == '-':
         lines.consume()
-        gerrit_orig_line += 1
-        gerrit_new_line += 1
-    return (gerrit_orig_line,
-            gerrit_new_line,
-            PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
-                                  side='b',
-                                  offset=(gerrit_new_line - lines.line_number())))
+        parser_state.gerrit_orig_line += 1
+        parser_state.gerrit_new_line += 1
+        parser_state.deleted_lines += 1
+    return PatchFileChunkLineMap(in_range=(in_start, lines.line_number() - 1),
+                                 side='b',
+                                 offset=(parser_state.gerrit_new_line - lines.line_number()))
 
 
 def _parse_patch_file_chunk(lines: InputSource,
-                            gerrit_orig_line: int,
-                            gerrit_new_line: int) -> Tuple[int, int, PatchFileChunkLineMap]:
+                            parser_state: HunkParserState) -> PatchFileChunkLineMap:
     line = lines[0]
     start_line_len = len(lines)
     if _does_match_end_of_super_chunk(lines):
         raise ValueError('Unexpected line: ' + line)
     elif line and line[0] == '+':
         logging.info('First char - 0: %c', line[0])
-        ret_val = _parse_patch_file_added_chunk(lines, gerrit_orig_line, gerrit_new_line)
+        ret_val = _parse_patch_file_added_chunk(lines, parser_state)
```

Comment:
nit: while we're here, can we update ret_val to something better?

It was called that since it was a tuple and didn't have a good name.
perhaps `chunk_map` ?